### PR TITLE
Implemented strongly typed internal function to run Cursorless commands

### LINKED ISSUE #1111: Switch to using strongly typed version of command internally
Instead of just using the vscode command for our manual tests, and possibly other call sites, we should call `commandRunner.runCommand`

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Fixes #1111

* Strongly typed command argument when calling internally
* Only a single constant in typescript and and another one in python that defines the Cursorless command identifier.

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [x] I have not broken the cheatsheet

### CODE REVIEW

File: src/core/commandRunner/CommandRunner.ts [hunk 1]
```diff
@@ -248,3 +250,7 @@ function constructThatTarget(
     return targets;
   }
 }
+
+export function runCursorlessCommand(command: Command) {
+  return vscode.commands.executeCommand(cursorlessCommand, command);
+}
```
Comment (pokey) [inline src/core/commandRunner/CommandRunner.ts:256]: Move to utility file

File: src/core/commandRunner/CommandRunner.ts [hunk 2]
```diff
@@ -20,6 +20,8 @@ import { ThatMark } from "../ThatMark";
 import { Command } from "./command.types";
 import { selectionToThatTarget } from "./selectionToThatTarget";
 
+const cursorlessCommand = "cursorless.command";
```
Comment (pokey) [inline src/core/commandRunner/CommandRunner.ts:23]: move out of this file

File: src/test/suite/crossCellsSetSelection.test.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import * as assert from "assert";
 import * as vscode from "vscode";
```
Comment (pokey) [inline src/test/suite/crossCellsSetSelection.test.ts:2]: unused import

File: src/test/suite/editNewCell.test.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import * as assert from "assert";
 import * as vscode from "vscode";
```
Comment (pokey) [inline src/test/suite/editNewCell.test.ts:2]: all of these

File: src/client-e2e-test/runCommand.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import * as vscode from "vscode";
+import { cursorlessCommandId } from "../common/commandIds";
+import { Command } from "../core/commandRunner/command.types";
```
Comment (pokey) [inline src/client-e2e-test/runCommand.ts:3]: This should eventually move into `common`, but maybe doesn't have to happen in this PR.  At the very least I'd do an `import type`

File: src/common/commandIds.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export const cursorlessCommandId = "cursorless.command";
```
Comment (pokey) [inline src/common/commandIds.ts:1]: Should this use `CONSTANT_CASE`?

File: src/test/suite/backwardCompatibility.test.ts [hunk 7]
```diff
@@ -19,14 +20,18 @@ async function runTest() {
...
 
-  await vscode.commands.executeCommand(
-    "cursorless.command",
-    "whatever",
-    "wrapWithPairedDelimiter",
-    [{ type: "primitive", selectionType: "line", mark: { type: "cursor" } }],
-    "(",
-    ")",
-  );
+  await runCursorlessCommand({
```
Comment (pokey) [inline src/test/suite/backwardCompatibility.test.ts:23]: This is a backward compatibility test.  Are you sure we want to upgrade it?

**Review by pokey — COMMENTED** [review]
Left a couple minor comments / questions; otherwise looks good

File: src/common/commandIds.ts (hunk 6 above)
Comment (AndreasArvidsson) [reply src/common/commandIds.ts:1]: If you like?

**Review by pokey — APPROVED** [review]

**Pull request merged**